packages/hooks/src/hooks/usePoller.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useRef } from "react";
+
+const usePoller = (func: () => void, delay: number): void => {
```

Comment:
We should attach some documentation to this function using `jsdoc` so users can understand its purpose:

```suggestion
/**
 * Calls a function at a set interval. The function will also be called
 * immediately.
 *
 * @param func The function to call at an interval
 * @param delay The delay between each interval call
 * @example
 * const callback = () => console.log('test')
 * usePoller(callback, 1000) // logs 'test' every second
 */
const usePoller = (func: () => void, delay: number): void => {
```